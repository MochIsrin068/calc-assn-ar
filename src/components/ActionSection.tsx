import React from "react";
import { Button } from "antd";

type TPropsActionSection = {
  onCheckResult: (actionType: string) => void;
};

export default function ActionSection({ onCheckResult }: TPropsActionSection) {
  return (
    <div className="my-4 flex items-center md:justify-center justify-between lg:justify-center pb-4 border-b-2">
      <Button
        type="primary"
        className="mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        size="large"
        onClick={() => onCheckResult("add")}
        title="Tambah"
        data-testid="add-button"
      >
        +
      </Button>
      <Button
        type="primary"
        className="mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        size="large"
        onClick={() => onCheckResult("substract")}
        title="Kurang"
        data-testid="substract-button"
      >
        -
      </Button>
      <Button
        type="primary"
        className="mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        size="large"
        onClick={() => onCheckResult("multiply")}
        title="Kali"
        data-testid="multiply-button"
      >
        X
      </Button>
      <Button
        type="primary"
        className="mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        size="large"
        onClick={() => onCheckResult("divide")}
        title="Bagi"
        data-testid="divide-button"
      >
        /
      </Button>
      <Button
        type="primary"
        className="bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        size="large"
        onClick={() => onCheckResult("reset")}
        title="Reset"
        data-testid="reset-button"
      >
        C
      </Button>
    </div>
  );
}
