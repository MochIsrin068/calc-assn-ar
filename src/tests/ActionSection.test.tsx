import React from "react";
import { render, screen } from "@testing-library/react";
import ActionSection from "../components/ActionSection";

const mockParams = {
  onCheckResult: jest.fn(),
};

describe("tests component ActionSection", () => {
  it("Perform snapshots test ActionSection", () => {
    const inputDefaultTree = render(<ActionSection {...mockParams} />);
    expect(inputDefaultTree).toMatchSnapshot();
  });

  it("test button action add exist and assertion title & value", () => {
    render(<ActionSection {...mockParams} />);
    const item = screen.queryByTestId("add-button");
    expect(item).toBeInTheDocument();
    expect(item?.title).toBe("Tambah");
    expect(item?.innerHTML).toBe("<span>+</span>");
  });

  it("test button action substract exist and assertion title & value", () => {
    render(<ActionSection {...mockParams} />);
    const item = screen.queryByTestId("substract-button");
    expect(item).toBeInTheDocument();
    expect(item?.title).toBe("Kurang");
    expect(item?.innerHTML).toBe("<span>-</span>");
  });

  it("test button action multiply exist and assertion title & value", () => {
    render(<ActionSection {...mockParams} />);
    const item = screen.queryByTestId("multiply-button");
    expect(item).toBeInTheDocument();
    expect(item?.title).toBe("Kali");
    expect(item?.innerHTML).toBe("<span>X</span>");
  });

  it("test button action divide exist and assertion title & value", () => {
    render(<ActionSection {...mockParams} />);
    const item = screen.queryByTestId("divide-button");
    expect(item).toBeInTheDocument();
    expect(item?.title).toBe("Bagi");
    expect(item?.innerHTML).toBe("<span>/</span>");
  });

  it("test button action reset exist and assertion title & value", () => {
    render(<ActionSection {...mockParams} />);
    const item = screen.queryByTestId("reset-button");
    expect(item).toBeInTheDocument();
    expect(item?.title).toBe("Reset");
    expect(item?.innerHTML).toBe("<span>C</span>");
  });
});
