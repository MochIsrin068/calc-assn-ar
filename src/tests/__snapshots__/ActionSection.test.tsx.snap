// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests component ActionSection Perform snapshots test ActionSection 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="my-4 flex items-center md:justify-center justify-between lg:justify-center pb-4 border-b-2"
      >
        <button
          class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
          data-testid="add-button"
          title="Tambah"
          type="button"
        >
          <span>
            +
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
          data-testid="substract-button"
          title="Kurang"
          type="button"
        >
          <span>
            -
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
          data-testid="multiply-button"
          title="Kali"
          type="button"
        >
          <span>
            X
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
          data-testid="divide-button"
          title="Bagi"
          type="button"
        >
          <span>
            /
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary ant-btn-lg bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
          data-testid="reset-button"
          title="Reset"
          type="button"
        >
          <span>
            C
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="my-4 flex items-center md:justify-center justify-between lg:justify-center pb-4 border-b-2"
    >
      <button
        class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        data-testid="add-button"
        title="Tambah"
        type="button"
      >
        <span>
          +
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        data-testid="substract-button"
        title="Kurang"
        type="button"
      >
        <span>
          -
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        data-testid="multiply-button"
        title="Kali"
        type="button"
      >
        <span>
          X
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg mr-3 bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        data-testid="divide-button"
        title="Bagi"
        type="button"
      >
        <span>
          /
        </span>
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg bg-blue-400 md:px-6 lg:px-6 px-5 py-4 text-xl flex items-center justify-center"
        data-testid="reset-button"
        title="Reset"
        type="button"
      >
        <span>
          C
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
